src/python/foglamp/device/server.py

```diff
@@ -126,5 +131,40 @@ def start(cls, plugin, core_mgt_host, core_mgt_port):
                 signal_name,
                 lambda: asyncio.ensure_future(cls._stop(loop)))
 
+        microservice_management_app = web.Application(middlewares=[middleware.error_middleware])
+        routes.setup(microservice_management_app)
+
+        microservice_management_handler = microservice_management_app.make_handler()
+        coro = loop.create_server(microservice_management_handler, '0.0.0.0', 0)
+        # added coroutine
+        microservice_management_server = loop.run_until_complete(coro)
+
+        microservice_management_address, microservice_management_port = microservice_management_server.sockets[0].getsockname()
+        _LOGGER.warning('Device - Management API started on http://%s:%s', microservice_management_address, microservice_management_port)
+
+        conn = http.client.HTTPConnection("{0}:{1}".format(core_mgt_host, core_mgt_port))
+        
+        service_registration_payload = {
+                "name"            : plugin,
+                "type"            : "Southbound",
+                "management_port" : int(microservice_management_port),
+                "service_port"    : 0,
+                "address"         : "127.0.0.1",
+                "protocol"        : "http"
+            }
+
+        conn.request(method='POST', url='/foglamp/service', body=json.dumps(service_registration_payload))
```

Comment:
move to a new `def register`? may be  we also can use `async with aiohttp.ClientSession() as session: ` aiohttp client